src/binary-reader-objdump.cc

```diff
@@ -57,12 +60,17 @@ class BinaryReaderObjdumpBase : public BinaryReaderNop {
   bool print_details = false;
   BinarySection reloc_section = BinarySection::Invalid;
   uint32_t section_starts[kBinarySectionCount];
+  bool section_found = false;
+  Stream* out_stream = nullptr;
 };
 
 BinaryReaderObjdumpBase::BinaryReaderObjdumpBase(const uint8_t* data,
                                                  size_t size,
                                                  ObjdumpOptions* options)
-    : options(options), data(data), size(size) {
+    : options(options),
+      data(data),
+      size(size),
+      out_stream(init_stdout_stream()) {
```

Comment:
Both the old code and the new code call init_stdout_stream() in the constructor of BinaryReaderObjdump.  In the old code the BinaryReaderObjdump is constructed up to 3 times (once for each of non-prepass phases).    In the new code it is constructed up to 2 times, since the BinaryReaderObjdumpDisassabler is not separated out.